post-scan-actions/aws-python-promote-or-quarantine/template.yml

```diff
@@ -87,46 +172,27 @@ Resources:
       MemorySize: 512
       Timeout: 30
       Tracing: Active
+      Role: !GetAtt LambdaExecutionRole.Arn
+      VpcConfig:
+        !If
+          - VpcEnabled
+          - SubnetIds:
+              - !Ref VpcSubnetId
+            SecurityGroupIds:
+              - !Ref VpcSecurityGroupId
+          - !Ref AWS::NoValue
       Environment:
         Variables:
           PROMOTEBUCKET: !Ref PromoteBucketName
           PROMOTEMODE: !Ref PromoteMode
           QUARANTINEBUCKET: !Ref QuarantineBucketName
           QUARANTINEMODE: !Ref QuarantineMode
+          KMSKEYARN: !Ref KMSKeyARN
           ACL: !Ref ACL
-      Policies:
-        - Statement:
-            Effect: Allow
-            Action:
-              - s3:GetObject
-              - s3:DeleteObject
-              - s3:GetObjectTagging
-            Resource:
-              - !Sub arn:${AWS::Partition}:s3:::${ScanningBucketName}/*
-        - Statement:
-            Effect: Allow
-            Action:
-              - s3:ListBucket
-            Resource:
-              - !Sub arn:${AWS::Partition}:s3:::${ScanningBucketName}
-        - Statement:
-            Effect: Allow
-            Action:
-              - s3:PutObject
-              - s3:PutObjectTagging
-              - !If
-                - PutACL
-                - s3:PutObjectAcl
-                - !Ref AWS::NoValue
-            Resource:
-              - !If
-                - PromoteEnabled
-                - !Sub arn:${AWS::Partition}:s3:::${PromoteBucketName}/*
-                - !Ref AWS::NoValue
-              - !If
-                - QuarantineEnabled
-                - !Sub arn:${AWS::Partition}:s3:::${QuarantineBucketName}/*
-                - !Ref AWS::NoValue
+      PermissionsBoundary: !If
+        - HasPermissionsBoundary
+        - !Ref PermissionsBoundary
+        - !Ref AWS::NoValue
```

Comment:
https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html#sam-function-permissionsboundary
According to doc: `This property works only if the role is generated for you.`, so the boundary should be configured in `LambdaExecutionRole` now.

Reply:
I will take a look at that...

Reply:
That should be fixed!

Reply:
This block would be redundant now.